src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileTargetLibrary.cs

```diff
@@ -8,80 +8,204 @@
 using NuGet.Shared;
 using NuGet.Versioning;
 
+#nullable enable
+
 namespace NuGet.ProjectModel
 {
     public class LockFileTargetLibrary : IEquatable<LockFileTargetLibrary>
     {
-        public string Name { get; set; }
+        #region Property packing
+
+        private readonly record struct PropertyKey(string PropertyName);
+
+        private static readonly PropertyKey DependenciesKey = new(nameof(Dependencies));
+        private static readonly PropertyKey FrameworkAssembliesKey = new(nameof(FrameworkAssemblies));
+        private static readonly PropertyKey FrameworkReferencesKey = new(nameof(FrameworkReferences));
+        private static readonly PropertyKey RuntimeAssembliesKey = new(nameof(RuntimeAssemblies));
+        private static readonly PropertyKey ResourceAssembliesKey = new(nameof(ResourceAssemblies));
+        private static readonly PropertyKey CompileTimeAssembliesKey = new(nameof(CompileTimeAssemblies));
+        private static readonly PropertyKey NativeLibrariesKey = new(nameof(NativeLibraries));
+        private static readonly PropertyKey BuildKey = new(nameof(Build));
+        private static readonly PropertyKey BuildMultiTargetingKey = new(nameof(BuildMultiTargeting));
+        private static readonly PropertyKey ContentFilesKey = new(nameof(ContentFiles));
+        private static readonly PropertyKey RuntimeTargetsKey = new(nameof(RuntimeTargets));
+        private static readonly PropertyKey ToolsAssembliesKey = new(nameof(ToolsAssemblies));
+        private static readonly PropertyKey EmbedAssembliesKey = new(nameof(EmbedAssemblies));
+        private static readonly PropertyKey PackageTypeKey = new(nameof(PackageType));
+
+        private readonly Dictionary<PropertyKey, object> _propertyValues = new();
```

Comment:
dictionary has some overhead. I know measuring an object is tricky, but I'm curious about the difference between before and after.